src/driver/alloy.rs

```diff
@@ -83,37 +94,37 @@ impl<Ext> BundleDriver<Ext> for EthCallBundle {
         trevm: crate::EvmNeedsTx<'a, Ext, Db>,
     ) -> DriveBundleResult<'a, Ext, Db, Self> {
         // 1. Check if the block we're in is valid for this bundle. Both must match
-        if trevm.inner().block().number.to() != self.block_number {
+        if trevm.inner().block().number.to::<u64>() != self.block_number {
             return Err(trevm.errored(BundleError::BlockNumberMismatch));
         }
 
         let bundle_filler = BundleBlockFiller::from(self.clone());
 
-        let trevm = trevm.with_block(
+        let run_result = trevm.try_with_block(
             |trevm| {
                 let mut trevm = trevm;
 
-                for raw_tx in self.txs.into_iter() {
+                for raw_tx in self.txs.clone().into_iter() {
                     let tx = TxEnvelope::decode(&mut raw_tx.to_vec().as_slice());
 
                     let tx = match tx {
                         Ok(tx) => tx,
-                        Err(e) => return trevm.errored(BundleError::TransactionDecodingError(e)),
+                        Err(e) => {
+                            return Err(trevm.errored(BundleError::TransactionDecodingError(e)))
+                        }
                     };
 
                     let run_result = trevm.run_tx(&tx);
 
                     match run_result {
                         // return immediately if errored
                         Err(e) => {
-                            return trevm.errored(BundleError::EVMError {
-                                inner: EVMError::Database(e.error()),
-                            })
+                            return Err(e.map_err(|e| BundleError::EVMError { inner: e }));
                         }
                         // Accept the state, and move on
                         Ok(res) => match res.result() {
                             ExecutionResult::Revert { .. } | ExecutionResult::Halt { .. } => {
-                                return trevm.errored(BundleError::BundleReverted);
+                                return Err(res.errored(BundleError::BundleReverted));
```

Comment:
the bundle contains a list of transactions that should be allowed to revert. we should check for membership in the list

Reply:
Note that this is `EthCallBundle`, which does not have this list